src/wgapi_userspace.rs

```diff
@@ -150,11 +150,18 @@ impl WireguardInterfaceApi for WireguardApiUserspace {
     fn assign_address(&self, address: &IpAddrMask) -> Result<(), WireguardInterfaceError> {
         debug!("Assigning address {address} to interface {}", self.ifname);
         let output = if cfg!(target_os = "macos") {
-            // On macOS, interface is point-to-point and requires a pair of addresses
-            let address_string = address.ip.to_string();
-            Command::new("ifconfig")
-                .args([&self.ifname, &address_string, &address_string])
-                .output()?
+            match address.ip {
+                IpAddr::V4(ip) => {
+                    let ip_string = ip.to_string();
```

Comment:
Please, move this above `match`, and use also in V6 variant. 🤓

Reply:
Done!  Thanks!